debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeTable.java

```diff
@@ -32,18 +37,21 @@
     private Lsn stopLsn;
 
     /**
-     * The table from which the changes are captured
+     * List of columns captured by the cdc table.
      */
-    private Table sourceTable;
+    private final List<String> capturedColumns;
 
-    public SqlServerChangeTable(TableId sourceTableId, String captureInstance, int changeTableObjectId, Lsn startLsn, Lsn stopLsn) {
+    public SqlServerChangeTable(TableId sourceTableId, String captureInstance, int changeTableObjectId, Lsn startLsn, Lsn stopLsn,
+                                String capturedColumnListString) {
         super(captureInstance, sourceTableId, resolveChangeTableId(sourceTableId, captureInstance), changeTableObjectId);
         this.startLsn = startLsn;
         this.stopLsn = stopLsn;
+        this.capturedColumns = Arrays.asList(BRACKET_PATTERN.matcher(Optional.ofNullable(capturedColumnListString).orElse(""))
```

Comment:
IMHO this should not be done in constructor but on the caller side. The string encoding is the stored procedure implementation detail and should not leak into logical representation of the capture table.

Reply:
Agreed, updated it